pilot/pkg/xds/gen2.go

```diff
@@ -88,6 +106,25 @@ func (s *DiscoveryServer) handleReqAck(con *XdsConnection, discReq *discovery.Di
 	return w, isAck
 }
 
+// ControlPlane identifies the instance and Istio version.
+func ControlPlane() *corev3.ControlPlane {
+	return controlPlane
+}
+
+func init() {
+	// The Pod Name (Pilot identity) is in PilotArgs, but not reachable globally nor from DiscoveryServer
+	podName := env.RegisterStringVar("POD_NAME", "", "").Get()
+	byVersion, err := json.Marshal(IstioControlPlaneInstance{
+		Component: "pilot",
```

Comment:
istiod - see above

Please include the clusterName - should be somewhere in mesh config or env. For multi-cluster.